File: 0162-find-peak-element/0162-find-peak-element.java
class Solution {
    public int findPeakElement(int[] nums) {
        int start  = 0;
        int end = nums.length-1;
        
        while(start<end){
            int mid = start + (end - start)/2;
            int mid2 = mid+1;
            
            if(nums[mid]<nums[mid2]){
                start = mid2;
            }else{
                end = mid;
            }
        }
        
        return start;
    }
}

File: 0162-find-peak-element/0162-find-peak-element.cpp
class Solution {
public:
    int search(vector<int> arr, int n){
        int low = 0;
        int high = n - 1;
        while(low<high){
            int mid = (low+high)/2;
            int mid2 = mid+1;
            if(arr[mid]<arr[mid2]){
                low = mid2;
            }else{
                high = mid;
            }
        }
        return low;
    }
    int findPeakElement(vector<int>& nums) {
        return search(nums, nums.size());
    }
};